app/models/framework/management_charge_calculator/sector_based.rb

```diff
@@ -0,0 +1,24 @@
+class Framework
+  module ManagementChargeCalculator
+    class SectorBased
+      attr_reader :central_government, :wider_public_sector
+
+      def initialize(central_government:, wider_public_sector:)
+        @central_government = central_government
+        @wider_public_sector = wider_public_sector
+      end
+
+      def calculate_for(entry)
+        (entry.total_value * (percentage(entry) / 100)).truncate(4)
+      end
+
+      private
+
+      def percentage(entry)
+        urn = entry.customer_urn
+        sector = Customer.find_by(urn: urn).sector
```

Comment:
We could use `pluck` here to avoid initialising a `Customer` instance. Will save on memory when performing the calculation for a large submission.

Reply:
Oh hold on, it’s an enumeration so that won’t work 😢

Reply:
I've been wondering about this, and wondering where the management charge call sites are (jobs?), and whether we can retrieve the `entry` with an associated `sector` projection in those circumstances.

Reply:
Yep, there is a job which loops through all entries, calculating and saving the charge on each. Feels like projecting that way would require that the job gain knowledge of the type of calculation, which would add additional complexity and coupling, so that cost would need to be weighed against any performance benefit. 

One other option keeping the existing boundaries might be to perform the lookup based on the enumeration here perhaps; the class API does expose the enumeration to us with Customer.sectors.

Reply:
We've used a `select` to at least limit the number of columns coming back to one. Happy to revisit this under general performance improvements later, but right now we have no onboarded frameworks with a sector-based management charge.